File: 3778-transform-array-by-parity/transform-array-by-parity.java
class Solution {
    public int[] transformArray(int[] nums) {
        int evenCount=0;
        int n=nums.length;
        int[]ans=new int[n];

        for (int num:nums){
           if (num % 2 == 0) {
                evenCount++;
            } 
        }
        for(int i=evenCount;i<n;i++){
            ans[i]=1;
        }
        return ans;
        
    }
}

File: 3778-transform-array-by-parity/transform-array-by-parity.py
class Solution(object):
    def transformArray(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        evenCount=0
        n=len(nums)
        ans=[0]*n
        for num in nums:
            if(num%2==0):
                evenCount+=1
        for i in range(evenCount,n):
            ans[i]=1
        return ans
        